src/isar/robot/robot.py

```diff
@@ -0,0 +1,261 @@
+import logging
+import time
+import queue
+from injector import inject
+from typing import Optional, Union, Any
+from isar.config.settings import settings
+from isar.models.communication.queues.queues import Queues
+from isar.models.communication.queues.status_queue import StatusQueue
+from isar.state_machine.states_enum import States
+from robot_interface.models.mission.status import RobotStatus
+from isar.services.utilities.threaded_request import (
+    ThreadedRequest,
+)
+from robot_interface.models.exceptions.robot_exceptions import (
+    ErrorMessage,
+    RobotCommunicationException,
+    RobotCommunicationTimeoutException,
+    RobotException,
+    RobotTaskStatusException,
+)
+from robot_interface.models.mission.status import TaskStatus
+from isar.models.communication.queues.queue_io import QueueIO
+from robot_interface.robot_interface import RobotInterface
+
+from robot_interface.models.mission.mission import Mission
+from robot_interface.models.mission.status import RobotStatus, TaskStatus
+from robot_interface.models.mission.task import Task
+from robot_interface.models.exceptions.robot_exceptions import (
+    ErrorMessage,
+    RobotException,
+    RobotInfeasibleTaskException,
+    RobotInfeasibleMissionException,
+)
+
+
+class Robot(object):
```

Comment:
Should this be split further so there is a separate class per thread? Similarly to how the telemetry is handled?

Reply:
I will leave this for a future refactor.